samples/RewriteSample/Startup.cs

```diff
@@ -14,8 +14,8 @@ public class Startup
         public void Configure(IApplicationBuilder app, IHostingEnvironment env)
         {
             var options = new RewriteOptions()
-                .AddRedirect("(.*)/$", "$1")
-                .AddRewrite(@"app/(\d+)", "app?id=$1", skipRemainingRules: false)
+                .AddRedirect("^/(.*)/$", "$1")
+                .AddRewrite(@"^/app/(\d+)", "app?id=$1", skipRemainingRules: false)
```

Comment:
Be super careful with the leading slashes. Double check both IIS and mod_rewrite for expectations on them, because IIRC they both explicitly disallow leading slashes.